### PR TITLE
Update fonts used by rustdoc

### PR DESCRIPTION
**Pull request by Manishearth**
Our version of Source Serif Pro is pretty old and is causing issues on Linux, see https://bugzilla.mozilla.org/show_bug.cgi?id=1545317 . I took this opportunity to update all of the fonts we use.

r? @steveklabnik @QuietMisdreavus

### CODE REVIEW

**Comment by Manishearth** [comment]
An example of how it's broken:

image

**Comment by Manishearth** [comment]
With the new fonts, on a test page:

image

File: src/librustdoc/html/static/FiraSans-LICENSE.txt [hunk 1]
```diff
@@ -24,7 +19,7 @@ with others.
 
 The OFL allows the licensed fonts to be used, studied, modified and
 redistributed freely as long as they are not sold by themselves. The
-fonts, including any derivative works, can be bundled, embedded,
+fonts, including any derivative works, can be bundled, embedded,
```
Comment (fbstj) [inline src/librustdoc/html/static/FiraSans-LICENSE.txt]: is the "random" trailing space to keep it inline with the original for some reason?

**Comment by Manishearth** [comment]
Yeah this is copied from the original

**Comment by QuietMisdreavus** [comment]
Something i wonder is whether this will affect docs.rs: Since we don't apply the resource suffix to the font files, docs.rs serves them directly (e.g. https://docs.rs/SourceSerifPro-Regular.ttf.woff). The updated files here are a drop-in upgrade of the fonts, right? So even if they start getting served for older docs, it won't degrade anything? (OTOH, i don't think it will actually serve them for a while, since the CDN will have cached the existing files.)

**Comment by Manishearth** [comment]
Yeah they're drop in upgrades. Shouldn't degrade anything.

**Review by QuietMisdreavus — APPROVED** [review]
This looks good, thanks!

**Pull request merged**